src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -853,6 +853,13 @@ trait Contexts { self: Analyzer =>
     private def isQualifyingImplicit(name: Name, sym: Symbol, pre: Type, imported: Boolean) =
       sym.isImplicit &&
       isAccessible(sym, pre) &&
+      !({
+        // [eed3si9n] ideally I'd like to do this: val fd = settings.isScala214 && sym.isDeprecated
+        // but implicit caching currently does not report sym.isDeprecated correctly.
+        val fd = settings.isScala214 && (sym == currentRun.runDefinitions.Predef_any2stringaddMethod)
+        if (settings.XlogImplicits && fd) echo(sym.pos, sym + " is not a valid implicit value because:\n" + "-Xsource:2.14 removes scala.Predef.any2stringadd")
```

Comment:
This logging is pretty noisy. When is intended to be shown?

```
$ qscalac -Xlog-implicits -Xsource:2.14 -d /tmp src/scalap/**/*.scala
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
src/scalap/scala/tools/scalap/Classfile.scala:122: materializing requested scala.reflect.type.ClassTag[String] using scala.reflect.`package`.materializeClassTag[String]()
      for ((x, i) <- entries.zipWithIndex ; if x != null) yield
                  ^
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
src/scalap/scala/tools/scalap/scalax/rules/scalasig/ClassFileParser.scala:233: materializing requested scala.reflect.type.ClassTag[Option[Any]] using scala.reflect.`package`.materializeClassTag[Option[Any]]()
  private val values = Array.fill[Option[Any]](size)(None)
                                                    ^
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
src/scalap/scala/tools/scalap/Decode.scala:57: refArrayOps is not a valid implicit value for bytes.type => ?{def take: ?} because:
inferred type arguments [Byte] do not conform to method refArrayOps's type parameter bounds [T <: AnyRef]
      bytes take length
      ^
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
src/scalap/scala/tools/scalap/Main.scala:109: materializing requested scala.reflect.type.ClassTag[String] using scala.reflect.`package`.materializeClassTag[String]()
        classname.split('.').map(NameTransformer.encode _).mkString(".")
                                ^
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
src/scalap/scala/tools/scalap/scalax/rules/Memoisable.scala:50: error: value + is not a member of AnyRef
      if(DefaultMemoisable.debug) println(key + " -> " + other)
                                              ^
src/scalap/scala/tools/scalap/scalax/rules/Memoisable.scala:56: error: value + is not a member of AnyRef
    if(DefaultMemoisable.debug) println(key + " -> " + t + " (" + out + ")")
                                            ^
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
src/scalap/scala/tools/scalap/scalax/rules/scalasig/ScalaSig.scala:33: wrapRefArray is not a valid implicit value for Array[Byte] => scala.collection.IterableOnce[?] because:
inferred type arguments [Byte] do not conform to method wrapRefArray's type parameter bounds [T <: AnyRef]
      case ConstValueIndex(index) => bytesForIndex(index)
                                                  ^
src/scalap/scala/tools/scalap/scalax/rules/scalasig/ScalaSig.scala:34: materializing requested scala.reflect.type.ClassTag[Byte] using scala.reflect.`package`.materializeClassTag[Byte]()
    }.toArray
      ^
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
src/scalap/scala/tools/scalap/scalax/rules/scalasig/ScalaSig.scala:45: refArrayOps is not a valid implicit value for bytes.type => ?{def take: ?} because:
inferred type arguments [Byte] do not conform to method refArrayOps's type parameter bounds [T <: AnyRef]
        ScalaSigAttributeParsers.parse(ByteCode(bytes.take(length)))
                                                ^
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
method any2stringadd is not a valid implicit value because:
-Xsource:2.14 removes scala.Predef.any2stringadd
two errors found
```

Reply:
I intended it to be called only when any2stringadd was invoked, but it seems like that's not the case.